examples/lock-app/efr32/src/LockManager.cpp

```diff
@@ -400,28 +413,42 @@ bool LockManager::setLockState(chip::EndpointId endpointId, DlLockState lockStat
     if (curState == lockState)
     {
         ChipLogDetail(Zcl, "Door Lock App: door is already locked, ignoring command to set lock state to \"%s\" [endpointId=%d]",
-                      lockStateToString(lockState), mEndpointId);
-        return false;
+                      lockStateToString(lockState), endpointId);
+        return true;
     }
 
+    // Check the RequirePINforRemoteOperation attribute
+    bool requirePin;
+    chip::app::Clusters::DoorLock::Attributes::RequirePINforRemoteOperation::Get(endpointId, &requirePin);
+
+    // If a pin code is not required
+    if (!requirePin)
+    {
+        ChipLogDetail(Zcl, "Door Lock App: setting door lock state to \"%s\" [endpointId=%d]", lockStateToString(lockState),
+                      endpointId);
+        curState = lockState;
+        return true;
+    }
+
+    // If a pin code is required but not given
     if (!pin.HasValue())
     {
-        ChipLogDetail(Zcl, "Door Lock App: PIN code is not specified, setting door lock state to \"%s\" [endpointId=%d]",
-                      lockStateToString(lockState), mEndpointId);
+        ChipLogDetail(Zcl, "Door Lock App: PIN code is not specified, but it is required [endpointId=%d]", mEndpointId);
         curState = lockState;
 
-        return true;
+        return false;
     }
 
     // Check the PIN code
-    for (uint8_t i; i < 10; i++)
+    for (uint8_t i = 0; i < mMaxCredentialsPerUser; i++)
     {
-        if (mLockCredentials.credentialType != DlCredentialType::kPin || mLockCredentials.status == DlCredentialStatus::kAvailable)
+        if (mLockCredentials[i].credentialType != DlCredentialType::kPin ||
+            mLockCredentials[i].status == DlCredentialStatus::kAvailable)
```

Comment:
Shouldn't this be looking at the credentials for the specific user, not just the ones at the front of the list?

Reply:
I changed the for loop to maxCredentialsPerUser